Augment country mask quality and switch to clipping; COUNTRY=rbd

ericboucher opened the pull request:
### Description

Improves print mask boundary quality via enhanced preprocessing in `preprocess-layers.ts`:

1. **Higher fidelity outlines** — filters tiny islands/holes (≥ 10 km² threshold), unions admin boundaries, then simplifies with tighter tolerance (`0.001`, high quality).
2. **Complex geometries** — Mozambique is always reprocessed with this pipeline. Indonesia uses a manually curated `admin-boundary-unified-polygon.json` (preprocessing skipped).
3. **Clipping** - Switch from masking to clipping.

Previously considered external boundaries (cartographyvectors.com) for cambodia/ecuador/mozambique/myanmar/sierraleone; those countries now rely on the improved preprocessing pipeline instead.

## How to test the feature:

- [ ] Open print/export for a small country (e.g. cambodia, jordan, zimbabwe) and verify mask outline is precise with no triangular artifacts
- [ ] Test mozambique print mask — no stripe artifacts
- [ ] Test indonesia print mask — boundary aligns with admin layers

## Checklist - did you ...

Test your changes with

- [ ] `REACT_APP_COUNTRY=rbd yarn start`
- [ ] `REACT_APP_COUNTRY=cambodia yarn start`
- [ ] `REACT_APP_COUNTRY=mozambique yarn start`
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

## Screenshot/video of feature:

ericboucher committed:
Update admin-boundary-unified-polygon.json

ericboucher committed:
Fix masks for complicated boundaries

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

wadhwamatic committed:
Merge branch 'master' into fix-cambodia-boundary

wadhwamatic commented:
@ericboucher - we can get admin0 files and add them during the config process going forward. Generally, the most reliable source is HDX, ideally datasets tagged with 'COD'. The format is usually going to be a shapefile.

https://data.humdata.org/dataset/?cod_level=cod-standard&cod_level=cod-enhanced&ext_administrative_divisions=1&q=&sort=last_modified%20desc&ext_page_size=25

I'll add Mozambique to this PR and revisit others soon.

wadhwamatic committed:
update mozambique boundary

wadhwamatic committed:
Revert "update mozambique boundary"

This reverts commit 3643d5b6f7b7cdd71ce9a31b797ab9efdb5e2ebb.

wadhwamatic commented:
Hmmm. When I added an admin0 boundary for Mozambique using the HDX data, the stripe returns. 
<img width="1786" alt="Screenshot 2024-02-28 at 14 55 59" src="https://github.com/WFP-VAM/prism-app/assets/3343536/c351cbd3-e565-4284-b316-d9257b970a42">

The issue with using the boundary you included @ericboucher is that they are not fully aligned with the other admin boundaries, as since this is for a print feature, fidelity matters. We shouldn't see boundaries like this:
<img width="1740" alt="Screenshot 2024-02-28 at 15 01 49" src="https://github.com/WFP-VAM/prism-app/assets/3343536/012dab09-20f8-4725-86e4-2093c47b8075">

wadhwamatic committed:
Merge branch 'master' into fix-cambodia-boundary

wadhwamatic committed:
revised moz adm0

wadhwamatic commented:
@ericboucher - the last moz boundary file I tried seems to work well. File size went up, but it's still under 2mb.

Here's what I did in QGIS:
1. Dissolve admin1 layers
2. Delete holes (this gets rids of small bodies of water)
3. Multipart to single part. This seems to connect strays
4. Delete unused features
5. Export geojson w/ precision 6

ericboucher committed:
Remove small areas to help reduce mask problems

ericboucher committed:
update script

wadhwamatic committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Merge master into fix-cambodia-boundary and address PR review

Port boundary preprocessing improvements into preprocess-layers.ts (turf v7,
area filtering, safer union/simplify). Resolve merge conflicts with master.
Remove unused wakatobi.json.json artifact.

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Update preprocess-layers.ts

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

wadhwamatic committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Map export - switch from mask to clip; country=indonesia (#1943)

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

Resolve conflicts in MapExportLayout (keep multi-map clip + React 19 paint scheduling), printConfig context (adminAreaClipPolygon + nullable printRef), and yarn.lock.

ericboucher committed:
Update printPreview.tsx

gislawill committed:
Replace export clip-path with source-level clipping. (#1951)

Clip raster tiles once via a custom clip:// MapLibre protocol and vector
layers at load with turf, so export/print uses a single map instead of
four stacked maps with per-frame CSS clip-path recomputation. WMS and
static raster sources route through clip:// when countryMask is active;
GeoJSON data layers are intersected to the admin polygon. Adds unit tests
for geometry classification, vector clipping, WMS clip wiring, and
single-map export layout.

ericboucher committed:
Reconcile clip reinforcements with source-level clipping (#1951).

Keep country-agnostic admin code matching and boundary preload for clip
polygon resolution; drop CSS clip-path and html2canvas bake now that export
uses ClipProvider and clip:// protocol on a single map.

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Fix issues

ericboucher committed:
Fix legend scale issue in batch map exports (#1952)

* Test legend scale

* Fixes

* Simplify legend scaling

ericboucher committed:
Simplify

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

wadhwamatic committed:
Merge branch 'master' into fix-cambodia-boundary

wadhwamatic committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Tentatively fix clip hangs

ericboucher committed:
empty

ericboucher committed:
Fix clip for non raster

ericboucher committed:
Filter admin areas

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary

ericboucher committed:
Additional layers clipped only when admin selected

ericboucher committed:
Simplify

ericboucher committed:
Merge branch 'master' into fix-cambodia-boundary
